data/src/test/java/org/apache/iceberg/io/TestWriterMetrics.java

```diff
@@ -146,4 +153,53 @@ public void verifySortedColMetric() throws Exception {
     Assert.assertFalse(upperBounds.containsKey(4));
     Assert.assertEquals(3L, (long) Conversions.fromByteBuffer(Types.LongType.get(), upperBounds.get(5)));
   }
+
+  @Test
+  public void testPositionDeleteMetrics() throws IOException {
+    Assume.assumeTrue(fileFormat == FileFormat.PARQUET);
+
+    WriterFactory<T> writerFactory = newWriterFactory(SCHEMA);
+    EncryptedOutputFile outputFile = fileFactory.newOutputFile();
+    PositionDeleteWriter<T> deleteWriter = writerFactory.newPositionDeleteWriter(outputFile, table.spec(), null);
+
+    try {
```

Comment:
Can we use try-with-resource

Reply:
I have to actually close the file before calling `toDeleteFile` and try-with-resource only allows me to reference the writer before it is closed. Matching the logic in the existing test in this class.

Reply:
Yep sorry, missed that